sklearn/ensemble/_gradient_boosting.pyx

```diff
@@ -203,6 +203,7 @@ def predict_stages(np.ndarray[object, ndim=2] estimators,
     cdef Tree tree
 
     if issparse(X):
+        assert X.format == 'csr'
```

Comment:
Can you use a ValueError to have a better error message? There is one error message below that maybe you can reuse.

```py
raise ValueError("X should be in np.ndarray or csr_matrix format,"
                              "got %s" % type(X))
```

Reply:
Also I guess `predict_stages` can be imported, so you could unit-test the error message.

Reply:
Fwiw I suggested an assert because this is an internal problem, not a user value error